apps/system/test/unit/nfc_handover_manager_test.js

```diff
@@ -408,27 +423,26 @@ suite('Nfc Handover Manager Functions', function() {
           requestId: 'req-id-1'
       };
 
-      spyBluetoothEnabledObserver = sinon.spy(function(event) {
-        MockBluetooth.enabled = event.settingValue;
-      });
-      MockNavigatorSettings.addObserver('bluetooth.enabled',
-                                        spyBluetoothEnabledObserver);
-
       spySendNDEF = sinon.spy(MockMozNfc.MockNFCPeer, 'sendNDEF');
       spySendFileViaHandover = sinon.spy(MockBluetoothTransfer,
         'sendFileViaHandover');
+
+      stubWriteSetting = this.sinon.stub(nfcHandoverManager, 'writeSetting',
```

Comment:
Don't stub writeSettings;
require mock_navigator_moz_settings and settings_core and test MockNavigatorMozSettings._settings['bluetooth.enabled'] is set

Refer airplane_mode_service_helper_test